src/server/protocol.ts

```diff
@@ -3772,44 +3129,30 @@ export const enum NewLineKind {
 }
 
 export const enum ScriptTarget {
-    ES3 = "ES3",
-    ES5 = "ES5",
-    ES6 = "ES6",
-    ES2015 = "ES2015",
-    ES2016 = "ES2016",
-    ES2017 = "ES2017",
-    ES2018 = "ES2018",
-    ES2019 = "ES2019",
-    ES2020 = "ES2020",
-    ES2021 = "ES2021",
-    ES2022 = "ES2022",
-    ESNext = "ESNext",
-}
-
-export const enum ClassificationType {
-    comment = 1,
-    identifier = 2,
-    keyword = 3,
-    numericLiteral = 4,
-    operator = 5,
-    stringLiteral = 6,
-    regularExpressionLiteral = 7,
-    whiteSpace = 8,
-    text = 9,
-    punctuation = 10,
-    className = 11,
-    enumName = 12,
-    interfaceName = 13,
-    moduleName = 14,
-    typeParameterName = 15,
-    typeAliasName = 16,
-    parameterName = 17,
-    docCommentTagName = 18,
-    jsxOpenTagName = 19,
-    jsxCloseTagName = 20,
-    jsxSelfClosingTagName = 21,
-    jsxAttribute = 22,
-    jsxText = 23,
-    jsxAttributeStringLiteralValue = 24,
-    bigintLiteral = 25,
+    ES3 = "es3",
+    ES5 = "es5",
+    ES6 = "es6",
+    ES2015 = "es2015",
+    ES2016 = "es2016",
+    ES2017 = "es2017",
+    ES2018 = "es2018",
+    ES2019 = "es2019",
+    ES2020 = "es2020",
+    ES2021 = "es2021",
+    ES2022 = "es2022",
+    ESNext = "esnext",
+    JSON = "json",
+    Latest = ESNext,
+}
+
+{
+    type AssertKeysComplete<Source extends { [K in keyof Target]: any; }, Target> = Source;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    type CopiedTypesComplete = [
+        AssertKeysComplete<typeof ModuleResolutionKind, typeof ts.ModuleResolutionKind>,
+        AssertKeysComplete<typeof ModuleKind, typeof ts.ModuleKind>,
+        AssertKeysComplete<typeof ScriptTarget, typeof ts.ScriptTarget>,
+        AssertKeysComplete<typeof JsxEmit, typeof ts.JsxEmit>,
+        AssertKeysComplete<typeof IndentStyle, typeof ts.IndentStyle>,
+    ];

```

Comment:
As mentioned before, what really matters is that the stringified enums represent every value for each option that the command line parser can recognize. Those strings aren’t encoded in types anywhere else that I know of, so rather than checking that directly, these checks just make sure each enum has all the keys the source enum has, protecting us from forgetting to update the protocol when compiler options get new values. The enum values here in the protocol could still be typo’d, as was the case for `"ReactNative"`